docs/draft-alakuijala-brotli-07.nroff

```diff
@@ -1008,9 +1008,10 @@ lengths and distances, respectively, is encoded in the meta-block
 header, and it must equal to the largest block type plus one in that
 block category. In other words, the set of literal, insert-and-copy
 length and distance block types must be [0..NBLTYPESL-1],
-[0..NBLTYPESI-1], and [0..NBLTYPESD-1], respectively. From this it
-follows that the alphabet size of literal, insert-and-copy length and
-distance block type codes is NBLTYPES + 2, NBLTYPESI + 2 and
+[0..NBLTYPESI-1], and [0..NBLTYPESD-1], respectively. If a block type
+has a value outside its respective range, the stream should be rejected as
+invalid. From this it follows that the alphabet size of literal, insert-and-copy
+length and distance block type codes is NBLTYPES + 2, NBLTYPESI + 2 and
```

Comment:
I had a whole thing written how it happened, because the stream can define its own symbols in a simple prefix tree.
When I now went back to the spec, I saw that it actually explicitly mentions:
> "If the integer value [for a symbol]
>   is greater than or equal to the alphabet size, then the stream
>   should be rejected as invalid."
I have missed that in my implementation thus far, and therefore it has been possible for a corrupt stream to create block type codes in a simple prefix code that exceed NBLTYPEx + 2.

I agree now that my suggested change is not needed. I will submit an amended change that fixes the type NBLTYPES -> NBLTYPESL.